### PR TITLE
MULE-9886: HttpResponseBuilder does not override Content-Length header after recalculating it

### PR DESCRIPTION
**Pull request by bayitajesi**

### CODE REVIEW

File: modules/http/src/main/java/org/mule/module/http/internal/listener/HttpResponseHeaderBuilder.java [hunk 1]
```diff
@@ -109,8 +109,9 @@ public void addContentType(String multipartFormData)
         addSimpleValue(HttpHeaders.Names.CONTENT_TYPE, multipartFormData);
     }
 
-    public void addContentLenght(String calculatedContentLenght)
+    public void setContentLenght(String calculatedContentLenght)
     {
+        removeHeader(HttpHeaders.Names.CONTENT_LENGTH);
```
Comment (afelisatti) [inline modules/http/src/main/java/org/mule/module/http/internal/listener/HttpResponseHeaderBuilder.java]: Static import.

File: modules/http/src/test/java/org/mule/module/http/internal/listener/HttpResponseBuilderTestCase.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    @Test
+    public void testContentLenghtIsOverridden() throws Exception
+    {
+        HttpResponseBuilder httpResponseBuilder = new HttpResponseBuilder();
+        mockMuleMessage(new ByteArrayInputStream(EXAMPLE_STRING.getBytes(StandardCharsets.UTF_8)), 12);
+
+        HttpResponse httpResponse = httpResponseBuilder.build(new org.mule.module.http.internal.domain.response.HttpResponseBuilder(), mockEvent);
+        Assert.assertEquals("13", httpResponse.getHeaderValue(CONTENT_LENGTH));
```
Comment (afelisatti) [inline modules/http/src/test/java/org/mule/module/http/internal/listener/HttpResponseBuilderTestCase.java]: Use hamcrest instead.

Comment (afelisatti) [inline modules/http/src/test/java/org/mule/module/http/internal/listener/HttpResponseBuilderTestCase.java]: Had a hard time understanding the idea of the test. Can you use the length of the String and comment somewhere that you are setting it wrong on purpose?

Comment (bayitajesi) [inline modules/http/src/test/java/org/mule/module/http/internal/listener/HttpResponseBuilderTestCase.java]: yes you're right, I'll change it

File: modules/http/src/test/java/org/mule/module/http/internal/listener/HttpResponseBuilderTestCase.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        mockEvent = mock(MuleEvent.class);
+        mockMuleMessage = mock(MuleMessage.class);
+    }
+
+    @Test
+    public void testContentLengthIsOverridden() throws Exception
+    {
+        HttpResponseBuilder httpResponseBuilder = new HttpResponseBuilder();
+        int contentLengthDifferingPayloadSize = 12;
+        mockMuleMessage(new ByteArrayInputStream(EXAMPLE_STRING.getBytes(StandardCharsets.UTF_8)), contentLengthDifferingPayloadSize);
```
Comment (afelisatti) [inline modules/http/src/test/java/org/mule/module/http/internal/listener/HttpResponseBuilderTestCase.java]: Static import.

File: modules/http/src/test/java/org/mule/module/http/internal/listener/HttpResponseBuilderTestCase.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+        HttpResponse httpResponse = httpResponseBuilder.build(new org.mule.module.http.internal.domain.response.HttpResponseBuilder(), mockEvent);
+        assertThat(httpResponse.getHeaderValue(CONTENT_LENGTH), is(String.valueOf(EXAMPLE_STRING.length())));
```
Comment (aiannucci) [inline modules/http/src/test/java/org/mule/module/http/internal/listener/HttpResponseBuilderTestCase.java]: Static import

File: modules/http/src/main/java/org/mule/module/http/internal/listener/HttpResponseBuilder.java [hunk 5]
```diff
@@ -261,7 +261,7 @@ private void setupContentLengthEncoding(HttpResponseHeaderBuilder httpResponseHe
             logger.debug("Content-Length encoding is being used so the 'Transfer-Encoding' header has been removed");
             httpResponseHeaderBuilder.removeHeader(TRANSFER_ENCODING);
         }
-        httpResponseHeaderBuilder.addContentLenght(String.valueOf(contentLength));
+        httpResponseHeaderBuilder.setContentLenght(String.valueOf(contentLength));
```
Comment (aiannucci) [inline modules/http/src/main/java/org/mule/module/http/internal/listener/HttpResponseBuilder.java]: Static import

**Pull request merged**